src/test/java/uk/gov/hmcts/reform/cwrdapi/service/impl/StaffRefDataServiceImplTest.java

```diff
@@ -336,6 +342,68 @@ void should_return_empty_list_of_case_worker_profile_with_status_code_200() {
         assertThat(searchResponse).isEmpty();
     }
 
+    @Test
+    void should_return_case_worker_profileSearch_with_status_code_200() {
+        ArrayList<CaseWorkerProfile> caseWorkerProfiles = new ArrayList<>();
+        CaseWorkerProfile caseWorkerProfile = buildCaseWorkerProfile();
+        caseWorkerProfiles.add(caseWorkerProfile);
+        Page<CaseWorkerProfile> pages = new PageImpl<>(caseWorkerProfiles);
+        searchReq = SearchRequest.builder()
+                .role(String.valueOf(1L))
+                .jobTitle(String.valueOf(1L))
+                .location("11112")
+                .serviceCode("SvcCode1")
+                .skill(String.valueOf(1L))
+                .userType("1")
+                .role("task supervisor,Staff Administrator")
+                .build();
+
+        when(caseWorkerProfileRepository.findByCaseWorkerProfiles(any(), any(), any(),
+                any(), any(), any(),any()))
+                .thenReturn(pages);
+        var pageRequest = validateAndBuildPagination(20, 1,
+                20, 1);
+
+        ResponseEntity<List<SearchStaffUserResponse>> responseEntity =
+                staffRefDataServiceImpl.retrieveStaffProfile(searchReq, pageRequest);
+        assertEquals(200, responseEntity.getStatusCodeValue());
+
+        assertThat(responseEntity.getHeaders().get("total-records").get(0)).isEqualTo("1");
+
+        List<SearchStaffUserResponse> searchResponse =
+                responseEntity.getBody();
+        assertThat(responseEntity.getBody()).isNotNull();
+        assertTrue(validateSearchUserProfileResponse(responseEntity,searchReq));
+    }
+
+
+    @Test
+    void should_return_empty_list_case_worker_profileSearch_with_status_code_200() {
+
+        var pageRequest =
+                validateAndBuildPagination(20, 1,
+                        20, 1);
+        ArrayList<CaseWorkerProfile> caseWorkerProfiles = new ArrayList<>();
+        Page<CaseWorkerProfile> pages = new PageImpl<>(caseWorkerProfiles);
+        searchReq = SearchRequest.builder()
+                .build();
+
+        when(caseWorkerProfileRepository.findByCaseWorkerProfiles(any(), any(), any(),
+                any(), any(), any(),any()))
+                .thenReturn(pages);
+        ResponseEntity<List<SearchStaffUserResponse>> responseEntity =
+                staffRefDataServiceImpl.retrieveStaffProfile(searchReq, pageRequest);
+        assertEquals(200, responseEntity.getStatusCodeValue());
+
+        assertThat(responseEntity.getHeaders().get("total-records").get(0)).isEqualTo("0");
+
+        List<SearchStaffUserResponse> searchResponse =
```

Comment:
unused variable searchResponse

Reply:
changes done

Reply:
@vasthav-patta , i can still see the unused variable searchResponse, pls check